### ISSUE TITLE
UCB suggestion: more sophisticated "issue tracking" system

### ISSUE THREAD

**Issue by jrwdunham** [OpenSourceFieldlinguistics/FieldDB#1943]
This is an interesting suggestion. Give users the ability to create issues (date- and creator-stamped), flag them as needing resolution, and then flag them as being resolved. The LingSync "status" field already fulfills much of this requirement. But it may be desirable to put more logic into this. That is, instead of creating conventions like "ToBeCheckedWithDave", we may want to do something more sophisticated. Something to think about anyways.

**Comment by cesine** [comment]
yes, i agree this is part of life in research teams. you can do @ username tags in comments which is how i ask my informants to review the data for x or y reason, and then when you login you search for @ yourusername to see if there is something you are supposed to do. 

![screen shot 2015-04-09 at 5 58 59 pm](https://cloud.githubusercontent.com/assets/196199/7065473/3298e80e-dee2-11e4-99b2-94d445e11ad8.png)

real issue tracking is a project in and of itself, and its unfortunately limitless. for example, do we want also to add a side system which is capable of emailing if someone replies to an issue, or asks the other to do something... it could be as comprehensive as re-inventing basecamp or github which is a bad idea :( I'd like to avoid any sort of emailing because that can be annoying

we already had one of the field methods instructors ask for a list of recent comments, and we have a map reduce for that. for sure, we can make a widget showing recent comments, with a filter input box on top which is by default to show the comments which contain the logged in user's name.

https://github.com/OpenSourceFieldlinguistics/FieldDB/blob/master/couchapp_dev/views/recent_comments/map.js#L51-55

my suggestion would be 
1. to make the widget for recent comments, 
2. with a search box on it pre-set to filter the comments to contain the username of who is logged in. and 
3. let the teams do what they end up doing with the comments as an open ended system for discussing/managing issues in the data. this is most likely how teams work on data in google docs, using comments...
4. add another operation or two to comments eg, "Resolved" where the comment is another color, and/or checked off, or hidden-ish. users can already update their comments and delete them which works for our TODOs so far, but this could be a even more lovely ..

you should probably tell them about the activity feeds too, they might like them if they are doing project management which is what normal research teams would be doing. 

btw, every element in the system has comments, so this means that pretty much every bit of data can have issue tracking features. the speadsheet only shows comments on the datum. for example data which needs to be cleaned, the comments go on the data list which contains the data etc

another option would be to add a new model to the system: issues, not re-use the comments.
